provider_example_test.go

```diff
@@ -0,0 +1,83 @@
+package tflog
+
+import (
+	"context"
+	"os"
+
+	"github.com/hashicorp/go-hclog"
+)
+
+func getExampleContext() context.Context {
+	return New(context.Background(), withOutput(os.Stdout),
+		WithLevel(hclog.Trace), WithoutLocation(), withoutTimestamp())
+}
+
+func ExampleWith() {
+	// context for example only; plugins should never need to do this
+	exampleCtx := getExampleContext()
```

Comment:
Admittedly, I was slightly confused when I saw this in the Go documentation the first time, as in, what is `getExampleContext()` since it is not shown and is not just `ctx := context.WithBackground()`? This was also prior to morning coffee and re-reading this it totally makes sense as-is having seen the implementation here.

Maybe this could be explained slightly differently by adjusting the comment wording slightly for these since it includes unexported calls in `New()`?

```suggestion
	// Example only logging context created with New().
	// Plugins will typically invoke New() directly, if not already
	// provided by an SDK.
	exampleCtx := getExampleContext()
```

Another option, probably less preferable, would be exporting `withOutput()` and `withoutTimestamp()` and showing the `New()` invocation directly saying that plugins should not really do those two things.

Totally fine to ignore this suggestion!

Reply:
I'll take a stab at rewording this to better frame expectations.